### PR TITLE
Add patterns list subcommand 

### LINKED ISSUE #2623: [mesheryctl] `pattern list`
Refer the [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0) for current status of `mesheryctl`.

#### Current Behavior
Related to #2266 and #2621, `mesheryctl` v0.5.0 supports the following `exp pattern` command operations:

```
Available Commands:
  apply       Apply pattern file
  delete      Delete pattern file
```

Meshery release v0.6.0 will support `pattern` as a stable command with `list` being one of the subcommands.

#### Desired Behavior
The following commands should be supported as well:

```
  list       Display list of all available pattern files

Flags:
   -a, --all    Display full length user and pattern file identifiers. 
```

### Implementation
1. Use `http://<current-context-endpoint>/api/experimental/patternfile` to retrieve a list of available patterns.
1. Require use of token to auth to endpoint.
1. Provide legible list of patterns in tabular format with optional `-a`, `--all` to show all pattern file metadata (not pattern itself).
1. Show total count of available patterns.

#### Acceptance Tests
1. `pattern list` command should be return the names of all available pattern
2. `pattern list` will require `-t` or `--token` has a valid token file as a flag.
3. `pattern list` will display the list of patterns in a tabular format. See table mockup.
3.a. Timestamps will be converted from `2021-03-23T20:28:50.229379Z` format to `MM/DD/YYYY HH:HH:HH`
3.b. User IDs will be truncated after first 8 characters.
4. `pattern list -a` will display the list of patterns in a tabular format with no truncation of details. See table mockup.

### Mockups
`list` output:

```
NAME                                    USER ID             CREATED          UPDATED
canary-rollout.yaml            090e7114...       01-13-2021       01-15-2021
split-traffic-50-50.yaml    090e7114...       03-23-2021     03-25-2021

Total: 2
```

`list -a` output:
```
PATTERN ID                                                               NAME                                   USER ID                                                                    CREATED      UPDATED
2a634cfe-87d5-478e-9fda-168580ff83ba     canary-rollout.yaml            090e7114-509a-4046-81f1-9c5fb8daf724    01-13-2021 08:20:10      01-15-2021 08:20:10
2a634cfe-87d5-478e-9fda-168580ff83ba     split-traffic-50-50.yaml    090e7114-509a-4046-81f1-9c5fb8daf724      03-23-2021 20:28:50     03-25-2021 20:28:50

Total: 2
```
Includes pattern id, full length ids, and hour:min:sec

---

#### Contributor Resources

- [mesheryctl Contributing Guide](https://github.com/layer5io/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)

### PR DESCRIPTION
**Pull request by s1ntaxe770r**
**Description**
Fixed  a few issues raised 
- signed commits 
- rename `--all` to `--verbose`
- fixed linting errors
This PR fixes #2623

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@s1ntaxe770r will you sync your fork so that the other commits are not included in this PR? You'll find instructions here - https://github.com/layer5io/meshery/blob/master/CONTRIBUTING-gitflow.md.

**Comment by s1ntaxe770r** [comment]
@leecalcote  would this do?

**Comment by leecalcote** [comment]
@s1ntaxe770r not quite. All of the other contributors commits above will disappear when the your fork and clone are synced.

File: mesheryctl/internal/cli/root/pattern/list.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+				PatternID := utils.TruncateID(v.ID.String())
+				var UserID string
+				if v.UserID != nil {
+					UserID = *v.UserID
+				} else {
+					UserID = "null"
+				}
+				PatterName := v.Name
+				CreatedAt := fmt.Sprintf("%d-%d-%d", int(v.CreatedAt.Month()), v.CreatedAt.Day(), v.CreatedAt.Year())
+				UpdatedAt := fmt.Sprintf("%d-%d-%d", int(v.UpdatedAt.Month()), v.UpdatedAt.Day(), v.UpdatedAt.Year())
```
Comment (aneeshnema) [inline mesheryctl/internal/cli/root/pattern/list.go:87]: date format should be `"%d-%d-%d %d:%d:%d"` for `verbose == true`

Also, I think you missed the case where `verbose == false` and `provider != "None"`

Comment (s1ntaxe770r) [reply mesheryctl/internal/cli/root/pattern/list.go]: great catch , I'll make changes

**Review by aneeshnema — APPROVED** [review]
LGMT 💯

**Comment by aneeshnema** [comment]
@s1ntaxe770r I think you have rebased incorrectly; that is why commits from other contributors are showing up.

**Review by navendu-pottekkat — CHANGES_REQUESTED** [review]
Looking through the code and it seems that you may have missed adding `list` to sub commands in `pattern.go`.

I did not test after that, will test it out after this change.

File: mesheryctl/internal/cli/root/pattern/list.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+)
+
+var (
+	token   string
+	verbose bool
+)
+
+var listCmd = &cobra.Command{
+	Use:  "list",
+	Long: "Display list of all available pattern files",
```
[the remark this answers is not in the document]
Comment (s1ntaxe770r) [reply mesheryctl/internal/cli/root/pattern/list.go]: oh snap, i totally forgot about this.

[the remark this answers is not in the document]
Comment (s1ntaxe770r) [reply mesheryctl/internal/cli/root/pattern/list.go]: fixed!

**Review by leecalcote — CHANGES_REQUESTED** [review]
Good stuff, @s1ntaxe770r. I think that all requested functionality and formatting is in here except for "Show total count of available patterns."

**Comment by s1ntaxe770r** [comment]
oof , almost forgot about that one , @leecalcote thoughts on showing the total count when the verbose flag is invoked?

**Comment by leecalcote** [comment]
No sweat. I think the idea here is that the total count will be show in any `list` output both without and with `-v`.

**Comment by s1ntaxe770r** [comment]
alright then

**Review by leecalcote — CHANGES_REQUESTED** [review]
@s1ntaxe770r you'll need to resync your fork.

**Comment by s1ntaxe770r** [comment]
done @leecalcote

**Comment by leecalcote** [comment]
Yay! 🎈

**Pull request merged**